lib/kafka/async_producer.rb

```diff
@@ -240,10 +244,23 @@ def run
       private
 
       def produce(*args)
-        @producer.produce(*args)
-      rescue BufferOverflow
-        deliver_messages
-        retry
+        retries = 0
+        begin
+          @producer.produce(*args)
+        rescue BufferOverflow => e
+          deliver_messages
+          if @max_retries == -1
+            retry
+          end
+          if retries < @max_retries
```

Comment:
Can you do `elsif` here instead, just to make the code a bit more legible?

Reply:
Sure. Thanks!